### ISSUE TITLE
Binwalk download fail

### ISSUE THREAD

**Issue by Maijin** [Zeecka/AperiSolve#10]
Hello o/

Binwalk output on the website is correct:

<img width="1249" alt="image" src="https://user-images.githubusercontent.com/1408600/99915743-cc439680-2d05-11eb-93d2-18492f4bc8d2.png">

Unfortunately the downloaded files aren't:

<img width="313" alt="image" src="https://user-images.githubusercontent.com/1408600/99915756-e1b8c080-2d05-11eb-98a6-6f2bdb1fd363.png">

When launching binwalk locally I got the proper files + filenames:

<img width="296" alt="image" src="https://user-images.githubusercontent.com/1408600/99915767-f1380980-2d05-11eb-9b31-1662cad647c2.png">

See also MD5, something definitely looks off:

<img width="483" alt="image" src="https://user-images.githubusercontent.com/1408600/99915787-0a40ba80-2d06-11eb-842e-811fa3b98f7e.png">

Reproducer can be found here:
[lol.zip](https://github.com/Zeecka/AperiSolve/files/5580220/lol.zip)

(Zsteg also looks stuck)

**Comment by Zeecka** [comment]
Thanks for reporting. I already knew this issue but it wasn't documented. I've got to dig into that.
For Zsteg it's because the process took too much time to complete. I think I'll rework the whole infrastructure to have microservices working on each task (and avoid Flask crashing). I should also keep a cache for recent submitted files (and maybe keep often submitted ones and make data analysis ?).

**Comment by Maijin** [comment]
Yeah that'd make sense if your tool start to be popular, it could suffer of a heavy use during CTFs over the same files.

Stats/data analysis is a good idea, depending what you want to do, you may not want to show the most recent info as it could leak flags of on-going CTFs, maybe add a delay of few days on the shown stats?

Could also add stats on the file submitted (like number of time submitted)

**Comment by Zeecka** [comment]
Update :
1. The original issue was due to options I set to binwalk. I usually set flag `-dd='.*'` to export data because it's more efficient. I assume you set flag `-e` that's why you got filenames. I'm working locally on the new version and I just set both flags ;).
2. I put some data analysis in this new version such as: passwords statistics, previous passwords submitted by other users for a given image, statistics about image (number of time submitted, ...) ...
3. In this new version, a daemon with multithreaded process is running to perform steganalysis. I'm working on an efficient cache for image that already have been submitted (but my storage is limited, so I've got to make a smart cache). I hope/think this new version will be faster.
4. No release date for the moment but I hope it'll be up until 2021. Backend progress = 80% (the garbage collector / cache is missing). Frontend progress = 50%.
